File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/Event.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner

/**
 * An event that describes a thing that happens
 */
data class Event(
    val id: String?,
    val categoryId: String?,
    val properties: List<EventProperty?>?,
    val status: EventStatus?,
    val createdBy: String?,
    val createdDateTime: String?,
    val active: Boolean?,
    val rating: Int?
) {

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        if (id != null) {
            joiner.add("id: " + GraphQLRequestSerializer.getEntry(id))
        }
        if (categoryId != null) {
            joiner.add("categoryId: " + GraphQLRequestSerializer.getEntry(categoryId))
        }
        if (properties != null) {
            joiner.add("properties: " + GraphQLRequestSerializer.getEntry(properties))
        }
        if (status != null) {
            joiner.add("status: " + GraphQLRequestSerializer.getEntry(status))
        }
        if (createdBy != null) {
            joiner.add("createdBy: " + GraphQLRequestSerializer.getEntry(createdBy))
        }
        if (createdDateTime != null) {
            joiner.add("createdDateTime: " + GraphQLRequestSerializer.getEntry(createdDateTime))
        }
        if (active != null) {
            joiner.add("active: " + GraphQLRequestSerializer.getEntry(active))
        }
        if (rating != null) {
            joiner.add("rating: " + GraphQLRequestSerializer.getEntry(rating))
        }
        return joiner.toString()
    }
}

File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/Event.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._
import EventStatus._

/**
 * An event that describes a thing that happens
 */
case class Event(
    id: String,
    categoryId: String,
    properties: scala.Seq[EventProperty],
    @com.fasterxml.jackson.module.scala.JsonScalaEnumeration(classOf[EventStatusTypeRefer])
    status: EventStatus,
    createdBy: String,
    createdDateTime: String,
    active: scala.Option[Boolean],
    rating: scala.Option[Int]
) {

    override def toString(): String = {
        scala.Seq(
            if (id != null) "id: " + GraphQLRequestSerializer.getEntry(id) else "",
            if (categoryId != null) "categoryId: " + GraphQLRequestSerializer.getEntry(categoryId) else "",
            if (properties != null) "properties: " + GraphQLRequestSerializer.getEntry(properties.asJava) else "",
            if (status != null) "status: " + GraphQLRequestSerializer.getEntry(status) else "",
            if (createdBy != null) "createdBy: " + GraphQLRequestSerializer.getEntry(createdBy) else "",
            if (createdDateTime != null) "createdDateTime: " + GraphQLRequestSerializer.getEntry(createdDateTime) else "",
            if (active.isDefined) "active: " + GraphQLRequestSerializer.getEntry(active.get) else "",
            if (rating.isDefined) "rating: " + GraphQLRequestSerializer.getEntry(rating.get) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }
}

